App.js

```diff
@@ -1,61 +1,103 @@
-  }
-
-  decrementValue = () => {
-    this.setState({
-      value: this.state.value - 1,
-      total_taps: this.state.total_taps + 1
-    })
-    console.log("Value: " + (this.state.value - 1))
-  }
-  
-  
-  render () {
-    return (
-      <View style={styles.container}>
-        <View>
-          <Image
-            source={{
-              uri: 'https://m.media-amazon.com/images/I/71uJRRJgeiL._AC_SS450_.jpg'
-            }}
-            style={{ width: 200, height: 200 }}
-            />
-        </View>
-        <Text style={{ fontSize: 60, marginBottom: -20}}>{this.state.value}</Text>
-        <Text style={{ fontSize: 12, padding: 20, color: 'grey'}}>{"Total taps: " + this.state.total_taps}</Text>
-        <StatusBar style="auto" />
-        <View style={{flexDirection:'row'}}>
-          <Button onPress={this.decrementValue} title="Decrease" />
-          <Text>   </Text>
-          <Button onPress={this.incrementValue} title="Increase" />
-        </View>
-      </View>
-    );
-  }
+//imports functions from different libraries
+import { StatusBar } from "expo-status-bar";
+import { useState } from "react";
+import { getNewColor } from "./getNewColor.js";
+import { Text, View, Button, Image } from "react-native";
 
-}
+//exports the function "app"
+export default function App() {
+  /*creates two constant variables and assigns them the setState function and a numeric value
+  so the numeric value can be dynamically changed later on.
+  */
+  const [value, setValue] = useState(0);
+  const [totalTaps, setTotalTaps] = useState(0);
+  const [color, setColor] = useState("#f0f8ff");
+
+  /* creates a variable and a function, and the values in the function are assigned to the variable.
+  The aforementioned dynamic variables are set to increment with 1 in this function.
+  */
+  const incrementValue = () => {
+    setValue(value + 1);
+    setTotalTaps(totalTaps + 1);
+
+    //this simply prints out what happens in this function.
+    console.log("Value: " + (value + 1));
+  };
+
+  //same as above
+  const decrementValue = () => {
+    setValue(value - 1);
+    setTotalTaps(totalTaps + 1);
+
+    console.log("Value: " + (value - 1));
+  };
+
+  /*similar as above but instead of incrementetion it uses a function to
+  change background colour.
+  */
+  const setBackgroundColor = () => {
+    /* function that gives you antiquewhite when you press aliceblue, then aliceblue
+    when you press antiquewhite
+    */
 
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    backgroundColor: '#fff',
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
-});
\ No newline at end of file
+    setColor(getNewColor(color));
+
+    console.log("Colour: " + color);
+  };
+
+  const styles = {
+    container: {
+      flex: 1,
+      backgroundColor: color,
+      alignItems: "center",
+      justifyContent: "center",
+    },
+  };
+
+  //the code that is returned to the script
+  return (
```

Comment:
Let's try to do some styling next?

I will try to repeat some explanations:
1. `<View></View>`, `<Image/>` etc. are called "React Components". We build a "Tree" of components to define, what should appear on the screen.
If the `/` is at the end of the element, like at `<Image/>`, that means that we can't put any other elements inside of the element.
Elements without `/`, like `<View>` are opening elements, when the `/` is at the start, like `</View>` they are closing elements.
We can put other elements between opening and closing elements to put them inside of the component.
When we for example write `<View><Image/></View>`, like here in our code, that means that the Image is inside of the View.
2. We can define components like this by ourselves, or import them from other places. In our case the components are imported from `react-native`. Here are the explanations for each component:
- `View`: https://reactnative.dev/docs/view <- This is basically just a box, where we can put other elements inside, and we can give it styling, like we do here with the background color.
- `Image`: https://reactnative.dev/docs/image
- `Text`: https://reactnative.dev/docs/text
- `Button`: https://reactnative.dev/docs/button
On these pages it has examples how to use it etc., and you can also play around in those sandboxes to understand them.
3. Components have "props" (short for "properties"), which we can add, like this `<View style={styles.container}>`. Here `style` is the property name, and `{styles.container}` is what we want that property to be.
In the React Native docs, you see how it lists all available props on the component pages: 
![image](https://user-images.githubusercontent.com/70113318/170063383-ca727efa-6909-4ba6-b2ae-b0ec90f48464.png)

Here are some more examples to the style prop, and the possible values we can put inside: https://reactnative.dev/docs/style

Lets try to do two things:
1. Go through those chapters under "Design" in https://reactnative.dev/docs/style and try to make our app look nicer, with defining heights etc., using Flexbox Layout, and setting colors.
2. In https://reactnative.dev/docs/button it says:
> A basic button component that should render nicely on any platform. Supports a minimal level of customization.
> If this button doesn't look right for your app, you can build your own button using [TouchableOpacity](https://reactnative.dev/docs/touchableopacity) or [TouchableWithoutFeedback](https://reactnative.dev/docs/touchablewithoutfeedback).

So we can't customize this button component so well. So let's replace the `Button` elements with the `TouchableOpacity` element and style it with the `style` property.

Reply:
-"We can put other elements between opening and closing elements to put them inside of the component.
When we for example write <View><Image/></View>, like here in our code, that means that the Image is inside of the View."

Actually didnt know this. I knew it worked kind of like this, but werent aware of how the system worked entirely so thats quite helpful actually.

-"Components have "props" (short for "properties"), which we can add, like this <View style={styles.container}>. Here style is the property name, and {styles.container} is what we want that property to be.
In the React Native docs, you see how it lists all available props on the component pages:"

This is actually very helpful as well. Had kind of a vague idea of how these concepts worked so this clarifies it a lot.